Dart/src/com/jetbrains/lang/dart/analyzer/DartServerData.java

```diff
@@ -577,11 +582,46 @@ public boolean isError() {
       return myCorrection;
     }
 
+    public @NotNull List<DartDiagnosticMessage> getContextMessages() {
+      return myContextMessages;
+    }
+
     public @Nullable @NonNls String getUrl() {
       return myUrl;
     }
   }
 
+  public static final class DartDiagnosticMessage extends DartRegion {
```

Comment:
There are 2 main reasons why we have classes like `DartRegion` and its inheritors instead of using classes from `org.dartlang.analysis.server.protocol.*` package directly:
- we need converted offsets and don't want to think about conversion each time we use offsets, so we store converted offsets in `DartRegion` object right away
- we need to update offsets/lengths on typing so that we have reasonable data before updated data comes from server

Neither is crucial for `DiagnosticMessage` so I updated the PR to store `org.dartlang.analysis.server.protocol.DiagnosticMessage` right in the `DartError` class, and removed `DartDiagnosticMessage` class.

Reply:
Thanks Alex, I thought I checked for the start line, but missed it.  Thanks for merging.